tools/configure/configure.ml

```diff
@@ -691,6 +688,27 @@ let operating_system =
   else
     (try Sys.getenv "OS" with Not_found -> "")
 
+(** Check for dune *)
+
+let dune_install_warning () =
+  warn "You are using Dune < 2.9, the install procedure will not respect the -docdir and -configdir configure directives; please see dev/doc/INSTALL.make.md for more information"
```

Comment:
I don't think we should silently succeed if Dune < 2.9 because of the issue described in #14833. I.e., we *need* the `-docdir` option (even if it's not used explicitly by the end-user) to set it to `share/doc` instead of just `doc`.

Reply:
OK
I care more about erroring if unsupported options are used tbh

Reply:
IIUC Dune could be configured at build time to use `share/doc` or `doc` and thus the Debian/Ubuntu packages might not have this issue. If it's possible to test if Dune will install the doc correctly, then we could avoid issuing the warning in that case.

Reply:
I was indeed checking was Debian is doing for doc, but indeed actually it seems they are just not installing any doc for ocaml packages.

The right fix for debian would be to patch Dune < 2.9 so docdir points to the right place.

Reply:
> if -docdir/-configdir were passed (or set interactively) and dune < 2.9, configure error

IMHO that's a bit overkill, a warning is fine, but we can do that if you want.

IMHO the main blocker is that in 8.13, Coq defaulted to `/usr/share/doc`, whereas Dune defaults to `/usr/doc` [opam layout], this is fixed now, and mainly concerns users installing manually.

How to go to a system where indeed Coq just generates the .install files and the rest is handled by distro infrastrucure is a good question.

Reply:
Patch for debian is pretty simple btw:
```diff
diff --git a/src/dune_rules/install.ml b/src/dune_rules/install.ml
index afa2acdad..9fd7e6779 100644
--- a/src/dune_rules/install.ml
+++ b/src/dune_rules/install.ml
@@ -209,7 +209,7 @@ module Section = struct
       let libexec_root = libdir in
       let share_root = Path.relative destdir "share" in
       let etc_root = Path.relative destdir "etc" in
-      let doc_root = Path.relative destdir "doc" in
+      let doc_root = Path.relative share_root "doc" in
       { lib_root
       ; libexec_root
       ; share_root
```
etc may need a similar patch too.

Reply:
>IMHO that's a bit overkill, a warning is fine, but we can do that if you want.

In general I feel warnings in configure are too easy to miss.

Reply:
>Patch for debian is pretty simple btw:

Could you send this patch to the Dune Debian package maintainer(s)?

Reply:
Done, see https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=993858